@navikt/core/react/src/form/combobox/Combobox.tsx

```diff
@@ -0,0 +1,247 @@
+  const mergedInputRef = useMemo(() => mergeRefs([inputRef, ref]), [ref]);
+
+  const { currentOption, toggleIsListOpen, filteredOptions } =
+    useFilteredOptionsContext();
+  const { customOptions, removeCustomOption, addCustomOption } =
+    useCustomOptionsContext();
+  const {
+    selectedOptions,
+    prevSelectedOptions,
+    removeSelectedOption,
+    addSelectedOption,
+    singleSelect,
+  } = useSelectedOptionsContext();
+
+  const {
+    hasError,
+    inputDescriptionId,
+    inputProps,
+    value,
+    size = "medium",
+    searchTerm,
+    setValue,
+    shouldAutocomplete,
+  } = useInputContext();
+
+  const focusInput = useCallback(() => {
+    inputRef.current?.focus?.();
+  }, []);
+
+  const handleClear = useCallback(
+    (event: ComboboxClearEvent) => {
+      onClear?.(event);
+      setValue("");
+    },
+    [setValue, onClear]
+  );
+
+  const handleDeleteSelectedOption = useCallback(
+    (clickedOption) => {
+      removeSelectedOption(clickedOption);
+      if (customOptions.includes(clickedOption))
+        removeCustomOption({ value: clickedOption });
+    },
+    [customOptions, removeCustomOption, removeSelectedOption]
+  );
+
+  const handleAddCustomOption = useCallback(
+    (event) => {
+      if (selectedOptions.includes(value.trim())) return;
+      addCustomOption({ value });
+      handleClear(event);
+      focusInput();
+    },
+    [selectedOptions, value, addCustomOption, handleClear, focusInput]
+  );
+
+  const toggleOption = useCallback(
+    (event) => {
+      const clickedOption = event?.target?.textContent;
+      const focusedOption = currentOption;
+      // onClick: toggle selected option
+      if (clickedOption) {
+        if (selectedOptions.includes(clickedOption)) {
+          handleDeleteSelectedOption(clickedOption);
+        } else if (filteredOptions.includes(clickedOption))
+          addSelectedOption(clickedOption);
+      }
+      // onEnter: remove selected filteredOption
+      else if (focusedOption && selectedOptions.includes(focusedOption)) {
+        removeSelectedOption(focusedOption);
+        if (customOptions.includes(focusedOption))
+          removeCustomOption({ value: focusedOption });
+      }
+      // onEnter: add focused option
+      else if (focusedOption && filteredOptions?.includes?.(focusedOption)) {
+        addSelectedOption(focusedOption);
+      }
+      //onEnter: add custom option
+      else if (focusedOption && !filteredOptions.includes(focusedOption)) {
+        handleAddCustomOption(event);
+      }
+    },
+    [
+      currentOption,
+      selectedOptions,
+      filteredOptions,
+      addSelectedOption,
+      handleDeleteSelectedOption,
+      removeSelectedOption,
+      customOptions,
+      removeCustomOption,
+      handleAddCustomOption,
+    ]
+  );
+
+  //focus on input whenever selectedOptions changes
+  useEffect(() => {
+    if (prevSelectedOptions !== selectedOptions) focusInput();
```

Comment:
Denne funksjonaliteten kan flyttes inn i `useSelectedOptionsContext` ([ref](https://react.dev/learn/you-might-not-need-an-effect))
```
const {
    selectedOptions,
    prevSelectedOptions,
    removeSelectedOption,
    addSelectedOption,
    singleSelect,
  } = useSelectedOptionsContext(inputRef);

```

Reply:
Done